File: framework/ecommerce-framework-api/src/main/java/com/sdl/ecommerce/api/model/ProductVariantAttributeValueType.java
package com.sdl.ecommerce.api.model;

/**
 * Product Variant Attribute Value Type
 *
 * @author nic
 */
public interface ProductVariantAttributeValueType {

    /**
     * @return value id
     */
    String getId();

    /**
     * @return value (can be localized)
     */
    String getValue();

    /**
     * @return if selected
     */
    boolean isSelected();

    /**
     * @return if applicable for current selection (e.g. a size which is not available for selected color)
     */
    boolean isApplicable();

}

File: dxa.net/ecommerce-framework-dotnet/SDL.ECommerce.OData/Model/ProductVariantAttributeValueType.cs
using SDL.ECommerce.Api.Model;

namespace SDL.ECommerce.OData
{
    public partial class ProductVariantAttributeValueType : IProductVariantAttributeValueType
    {
        bool IProductVariantAttributeValueType.IsSelected
        {
            get
            {
                return (bool) this.IsSelected;
            }
        }

        bool IProductVariantAttributeValueType.IsAvailable
        {
            get
            {
                return (bool)this.IsApplicable;
            }
        }

    }
} 
